snap/snapcraft.yaml

```diff
@@ -0,0 +1,58 @@
+name: yaru-icons-example
+version: git
+summary: Yaru Icons
+description: Yaru Icons Example
+confinement: strict
+base: core22
+grade: stable
+icon: snap/gui/icon.svg
+architectures:
+  - build-on: amd64
+  - build-on: arm64
+
+parts:
+  flutter-git:
+    source: https://github.com/flutter/flutter.git
+    source-branch: stable
+    source-depth: 1
+    plugin: nil
+    override-build: |
+      mkdir -p $CRAFT_PART_INSTALL/usr/bin
+      mkdir -p $CRAFT_PART_INSTALL/usr/libexec
+      cp -r $CRAFT_PART_SRC $CRAFT_PART_INSTALL/usr/libexec/flutter
+      ln -s $CRAFT_PART_INSTALL/usr/libexec/flutter/bin/flutter $CRAFT_PART_INSTALL/usr/bin/flutter
+      ln -s $CRAFT_PART_INSTALL/usr/libexec/flutter/bin/dart $CRAFT_PART_INSTALL/usr/bin/dart
+    build-packages:
+      - clang
+      - cmake
+      - curl
+      - ninja-build
+      - unzip
+      - xz-utils
+      - zip
+    override-prime: ""
+
+  yaru-icons-example:
+    after: [flutter-git]
+    plugin: nil
+    source: .
+    override-build: |
+      set -eux
+      cd example
+      flutter channel stable
+      flutter upgrade
+      flutter doctor
+      dart pub get
+      flutter build linux --release -v
+      mkdir -p $CRAFT_PART_INSTALL/bin/
+      cp -r build/linux/*/release/bundle/* $CRAFT_PART_INSTALL/bin/
+
+apps:
+  yaru-icons-example:
+    command: bin/yaru_icons_example
+    desktop: snap/gui/yaru-icons-example.desktop
+    extensions: [gnome]
+    plugs:
+      - gsettings
+      - network
+      - network-manager-observe
```

Comment:
does the example need network?

Reply:
Not at all, indeed 😅

Reply:
ok, let's leave these two lines out. otherwise looks good to me

Reply:
Done, thanks for the review :)